src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -892,21 +979,23 @@ private static ReadOnlySpan<char> EatAllWhitespace(ReadOnlySpan<char> str)
             // Loop through the remaining chars, copying over non-whitespace.
             for (; i < str.Length; i++)
             {
-                char c = str[i];
-                if (!char.IsWhiteSpace(c))
+                TChar c = str[i];
+                if (!IsWhite(TChar.CastToUInt32(c)))
                 {
                     chArr[newLength++] = c;
                 }
             }
 
             // Return the string with the whitespace removed.
-            return new ReadOnlySpan<char>(chArr, 0, newLength);
+            return new ReadOnlySpan<TChar>(chArr, 0, newLength);
         }
 
-        private static bool IsHexPrefix(ReadOnlySpan<char> str, int i) =>
+        private static bool IsHexPrefix<TChar>(ReadOnlySpan<TChar> str, int i) where TChar : unmanaged, IUtfChar<TChar> =>
             i + 1 < str.Length &&
-            str[i] == '0' &&
-            (str[i + 1] | 0x20) == 'x';
+            str[i] == TChar.CastFrom('0') &&
+            (str[i + 1] | TChar.CastFrom(0x20)) == TChar.CastFrom('x');
+
+        private static bool IsWhite(uint ch) => (ch == 0x20) || ((ch - 0x09) <= (0x0D - 0x09));
```

Comment:
This isn't the same with `char.IsWhiteSpace`.

Reply:
Is there something I should use instead for TChar? I copied this from https://github.com/dotnet/runtime/blob/ee3b1f0d5ba90f0cdbf04a59c0624fc87fb2cc8b/src/libraries/Common/src/System/Number.Parsing.Common.cs#L309  It's used there and in Number.Parsing.cs for removing whitespace from TChar spans.

Reply:
It's not that easy to have `IsWhite(TChar)`, since UTF8 is operated in `Rune`.
In this case, you'd have to write different code for UTF8 and UTF16, guarded by `if (typeof(TChar) == typeof(byte))`, then use `Unsafe.BitCast` to cast from `Span<TChar>` to `Span<byte>`.
For UTF8 code, you should enumerate the original span in runes. You can follow the implementation of `TrimUtf8`.